src/api/todo.ts

```diff
@@ -0,0 +1,90 @@
+import { Todo } from '../types/Todo';
+
+export const getTodos = (): Todo[] => {
+  const todos = getTodoInLocalStorage();
+  if (todos === null) {
+    return [];
+  }
+  return JSON.parse(todos);
+};
+
+export const createTodo = (content: string): Todo => {
+  const todos = getTodos();
+  const todo = {
+    content,
+    id: new Date().getTime(),
+    completed: false,
+    createDatetime: new Date().toString(),
+  };
+  todos.push(todo);
+  setTodoInLocalStorage(todos);
+  console.info(`✅ 뚜두 등록 성공`);
+  return todo;
+};
+
+export const updateTodo = (_todo: Todo) => {
+  const todos = getTodos();
+  const updatedTodos = todos.map(todo => {
+    if (todo.id !== _todo.id) {
+      return todo;
+    }
+    return {
+      ..._todo,
+      updateDatetime: new Date().toString(),
+    };
+  });
+  setTodoInLocalStorage(updatedTodos);
+  console.info(`✅ 뚜두 변경 성공`);
+};
+
+export const toggleTodo = ({
+  id,
+  completed,
+}: {
+  id: number;
+  completed: boolean;
+}) => {
+  const todos = getTodos();
+  const toggledTodos = mapToggledTodos({ id, completed, todos });
+  setTodoInLocalStorage(toggledTodos);
+};
+
+export const deleteTodo = (_id: number) => {
+  const todos = getTodos();
+  const deletedTodos = todos.filter(todo => todo.id !== _id);
+  setTodoInLocalStorage(deletedTodos);
+  console.info(`✅ 뚜두 삭제 성공`);
+};
+
+const setTodoInLocalStorage = (todos: Todo[]) => {
+  console.info(`✅ LocalStorage에 저장되는 Todos : ${JSON.stringify(todos)}`);
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+const getTodoInLocalStorage = (): string | null => {
+  return localStorage.getItem('todos');
+};
+
+const mapToggledTodos = ({
+  id,
+  completed,
+  todos,
+}: {
+  id: number;
+  completed: boolean;
+  todos: Todo[];
+}): Todo[] => {
+  return todos.map(todo => {
```

Comment:
todo 라는 파라미터의 값을 직접적으로 변환하는 건 별로 좋은 패턴이 아니라고 알고 있어요. ([이 페이지](https://wikidocs.net/605)나 'remove assignments to parameters' 로 검색하면 관련 글을 볼 수 있습니다.)

`temp = todo;` 이런식으로 변수를 쪼개서 temp 라는 변수의 값을 사용하셔도 좋고, 참고만 하셔도 좋은데 저는 주로 아래와 같이 씁니다. 혹은 if 문을 삼항연산자로 바꿔서 쓰기도 합니다.

```suggestion
  return todos.map(todo => {
    if (todo.id !== id) return todo;
    return {
      ...todo,
      completed,
      updateDatetime: new Date().toString(),
      completedDatetime: completed ?? new Date().toString(),
    }
  })
```

Reply:
삼항 연산자 부분은 생각은 했었는데 지난 56개의 PR에서 가독성이 떨어진다는 리뷰를 받아서 사용하지 않았습니다!
저도 상당히 편하다고 생각하는데 아쉽더라구용

Reply:
해당 직접 파라미터를 고치는 부분은 느낌적으로만 알고 있었는데 이번에 확실해 알게 되었어요 감사해용!
@Leehaeun0 

ps. 삼항연산 안쓰니까 펼치기 문법쓰면 코드가 너무 들어워져서 직접 접근으로 했는데 오히려 확실하게 알게되는게기가 되었습니당!!!

Reply:
@Leehaeun0 해결~